homeassistant/components/sensor/foobot.py

```diff
@@ -0,0 +1,147 @@
+PARALLEL_UPDATES = 1
+
+TIMEOUT = 10
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_TOKEN): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the devices associated with the account."""
+    from foobot_async import FoobotClient
+
+    token = config.get(CONF_TOKEN)
+    username = config.get(CONF_USERNAME)
+
+    client = FoobotClient(token, username,
+                          async_get_clientsession(hass),
+                          timeout=TIMEOUT)
+    data = {}
+    dev = []
+    try:
+        devices = yield from client.get_devices()
+        _LOGGER.debug("The following devices were found: %s", devices)
+        for device in devices:
+            for sensor_type in SENSOR_TYPES.keys():
+                if sensor_type != 'time':
+                    foobot_sensor = FoobotSensor(client, data, device,
+                                                 sensor_type)
+                    dev.append(foobot_sensor)
+    except (aiohttp.client_exceptions.ClientConnectorError,
+            asyncio.TimeoutError, client.ClientError):
+        _LOGGER.exception('Failed to connect to foobot servers.')
+        raise PlatformNotReady
+    async_add_devices(dev, True)
+
+
+class FoobotSensor(Entity):
+    """Implementation of a Foobot sensor."""
+
+    def __init__(self, client, data, device, sensor_type):
+        """Initialize the sensor."""
+        self._client = client
+        self._uuid = device['uuid']
+        self._name = 'Foobot {} {}'.format(device['name'],
+                                           SENSOR_TYPES[sensor_type][0])
+        self.type = sensor_type
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._last_request = datetime.utcfromtimestamp(0)
+        self._data = data
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return SENSOR_TYPES[self.type][2]
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        try:
+            data = self._data[self._uuid][self.type]
+        except(KeyError, TypeError):
+            data = None
+        return data
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Get the latest data."""
+        if self._uuid in self._data:
+            _LOGGER.debug('last_update: %s', self._last_request)
+            _LOGGER.debug('now: %s', datetime.utcnow())
+            if self._last_request + SCAN_INTERVAL > datetime.utcnow():
+                # If the last request made is recent enough,
+                # just skip the update
+                return
```

Comment:
Don't know whether this is possible, but maybe have a look at the `Throttle` util class:

https://github.com/home-assistant/home-assistant/blob/ad8fe8a93a45584515b36611f89feba336ea7816/homeassistant/util/__init__.py#L238-L243

Reply:
Unfortunately, it works only for limiting call inside an instance, not between multiple instances, I changed to make sure that all sensors devices uses the same data cache.